lightly/api/api_workflow_tags.py

```diff
@@ -73,3 +73,75 @@ def get_filenames_in_tag(
             masked_select_from_list(filenames_on_server)
 
         return filenames_tag
+
+    def create_tag_from_filenames(
+        self,
+        list_of_filenames: List[str],
+        new_tag_name: str,
+        parent_tag_id: str = None
+    ) -> TagData:
+        """Creates a new tag from a list of filenames.
+
+        Args:
+            list_of_filenames:
+                A list of filenames to be included in the new tag.
+            new_tag_name:
+                The name of the new tag.
+            parent_tag_id:
+                The tag defining where to sample from, default: None resolves to the initial-tag.
+
+        Returns:
+            The newly created tag.
+
+        Raises:
+            RuntimeError
+        """
+        
+        # make sure the tag name does not exist yet
+        tags = self.get_all_tags()
+        if new_tag_name in [tag.name for tag in tags]:
+            raise RuntimeError(f'There already exists a tag with tag_name {new_tag_name}.')
+        if len(tags) == 0:
+            raise RuntimeError('There exists no initial-tag for this dataset.')
+
+        # fallback to initial tag if no parent tag is provided
+        if parent_tag_id is None:
+            parent_tag_id = tags[-1].id
+
+        tot_size = tags[-1].tot_size
+
+        # get list of filenames from tag
+        fnames_tag = self.get_filenames_in_tag(tags[-1])
+
+        # create new bitmask 
+        bitmask = BitMask(tot_size)
```

Comment:
This does not seem the correct way of creating a bitmask, see the docs.
Rather use `Bitmask.from_length(tot_size)`. This is initialised by 1.
 
 I would rather do
 ```
 filenames_server = self.get_filenames()
 
bitmask = BitMask.from_length(tot_size)
filenames_new_tag = set(filenames_new_tag)
for i, fname in enumerate(filenames_server):
    if fname not in filenames_new_tag:
        bitmask.unset_kth_bit(i)
```
This also removes the squared complexity

Reply:
good suggestion. For what its worth: ` if fname not in filenames_new_tag:` would always search till the end and not terminate early causing it to be n*m instead of n*m/2

Reply:
Hmmm, that was intentional. The fastest way to create a bitmask is the normal init. You have to loop through both arrays anyway. Not sure how you can go sub "quadratic" or NxM as Jeremy pointed it out :) 
I did rewrite the part though to not use a for loop but a simple `if .. in ..` check.

Reply:
@MalteEbner suggests using a set of filenames which reduces the lookup speed to "constant" time. Cleanest solution would probably be to create an index of filename -> index in linear time and then create an empty bitmask and iterate over the filenames in the tag and set the bit at the index.

Reply:
Yes, the solution I proposed is linear, because the set lookup is constant and not linear.

Reply:
Furthermore, `Bitmask(x)` is still unequal to `Bitmask.from_length(x)`. However, the initial length does not matter, as the bits are unset anyway. You can just use` Bitmask(0)` instead, this makes it clear that it does not matter what is inserted during the initialization.

Thus my proposal can be made easier to

```
filenames_server = self.get_filenames()

bitmask = BitMask(0) //initialize empty bitmask
filenames_new_tag = set(filenames_new_tag)
for i, fname in enumerate(filenames_server):
   if fname in filenames_new_tag: # constant time, as it is a set lookup.
       bitmask.set_kth_bit(i)
```

Reply:
I see, you're right. I missed the `set(...)` part :)